mindmeld/components/request.py

```diff
@@ -13,12 +13,62 @@
 import logging
 import attr
 import immutables
+import iso639
 from pytz import timezone
 from pytz.exceptions import UnknownTimeZoneError
 
+from ..constants import MINDMELD_LOCALES
+
 logger = logging.getLogger(__name__)
 
 
+def _validate_language_code(param=None):
+    """Validates language code parameters
+
+    Args:
+        param (str, optional): The time zone parameter
+
+    Returns:
+        str: The passed in time zone
+    """
+    if not param:
+        return None
+    if not isinstance(param, str):
+        logger.error("Invalid %r param: %s is not of type %s.", 'language', param, str)
+        return None
+    try:
+        # Make the string uppercase by default since that is the character format accepted by
+        # the system entity resolver
+        iso639.to_name(param.upper())
+    except iso639.NonExistentLanguageError:
+        logger.error("Invalid %r param: %s is not of type %s.", 'language', param, str)
+        return None
+    return param.upper()
+
+
+def _validate_locale_code(param=None):
+    """Validates the locale code parameters
+
+    Args:
+        param (str, optional): The time zone parameter
+
+    Returns:
+        str: The passed in time zone
+    """
+    if not param:
+        return None
+    if not isinstance(param, str):
+        logger.error("Invalid %r param: %s is not of type %s.", 'locale', param, str)
+        return None
+
+    if param in MINDMELD_LOCALES:
+        return param
+    else:
+        logger.error("Invalid %r param: %s is not of valid MindMeld locale which include: %s",
+                     'locale', param, MINDMELD_LOCALES)
+        return None
```

Comment:
We could check that both parts of the locale are valid codes, but it's not necessary to check that the combination is known.